0_2_pandas_dataframes_attributes_methods.Rmd

```diff
@@ -533,6 +602,65 @@ cols = ['Fertility Rate', 'Human Development Index']
 # Select columns by direct indexing, then describe.
 df[cols].describe()
 ```
+::: {exercise-start}
+:label: by-any-means
+:class: dropdown
+:::
+
+Can you think of what code you need to write to create a Series that contains the just the means of the `Human Development Index`, `Fertility Rate` and `Population` variables? For good measure, lets also require that the `name` attribute is set to `mean`. The output should look like this:
+
+```
+Human Development Index      0.789333
+Fertility Rate               1.773867
+Population                 236.719867
+Name: mean, dtype: float64
+```
+
+Try to do this in as few lines of code as possible, using the methods shown on this page...
+
+```{python}
+# your code here
+```
+
+::: {exercise-end}
+:::
+
+::: {solution-start} by-any-means
+:class: dropdown
+:::
+
+The output of the `.describe()` method is itself a Dataframe, and the `index` of this Dataframe contains a row called `mean`, which, shockingly, contains the means. As a result, you can use `.loc` indexing to retrieve just the means from the output of `.describe()`:
```

Comment:
Maybe a hint here about `.describe` having the useful property that, by default, it drops non-numeric variables.